spring-integration-file/src/main/java/org/springframework/integration/file/FileReadingMessageSource.java

```diff
@@ -334,8 +334,10 @@ protected void onInit() {
 			this.scanner = new WatchServiceDirectoryScanner();
 		}
 
+		// Check that the local filter and locker options are _NOT_ set if an external scanner has been set.
+		// The external scanner is responsible for the filter and locker options in that case.
 		Assert.state(!(this.scannerExplicitlySet && (this.filter != null || this.locker != null)),
-				"The 'filter' and 'locker' options must be present on the provided external 'scanner': "
+				"When using an external scanner the local 'filter' and 'locker' options should not be used. Instead, set these options on the external DirectoryScanner: "
```

Comment:
I think `local` word here and in the comment above is misleading.
This `FileReadingMessageSource` definitely works only with the local file system.

Also, please, add your name to `@author`